spotinst/ocean_aws_launch_configuration/fields_spotinst_ocean_aws_launch_configuration.go

```diff
@@ -701,10 +702,12 @@ func expandInstanceMetadataOptions(data interface{}) (*aws.InstanceMetadataOptio
 	if v, ok := m[string(HTTPTokens)].(string); ok && v != "" {
 		instanceMetadataOptions.SetHTTPTokens(spotinst.String(v))
 	}
-	if v, ok := m[string(HTTPPutResponseHopLimit)].(int); ok && v >= 0 {
-		instanceMetadataOptions.SetHTTPPutResponseHopLimit(spotinst.Int(v))
-	} else {
-		instanceMetadataOptions.SetHTTPPutResponseHopLimit(nil)
+	if v, ok := m[string(HTTPPutResponseHopLimit)].(int); ok {
```

Comment:
Can we add a comment why this default value is used?

Reply:
The default value mentioned is used to set HTTPPutResponseHopLimit field to null when the customer doesn't want to set this param, as terraform set it 0 for integer type param by default.